tests/backup/backup_helper.go

```diff
@@ -4171,3 +4159,105 @@ func CreateRuleForBackupWithMultipleApplications(orgID string, appList []string,
 	}
 	return preRuleName, postRuleName, nil
 }
+
+// GetAllBackupNamesByOwnerID gets all backup names associated with the given ownerID
+func GetAllBackupNamesByOwnerID(ownerID string, orgID string, ctx context.Context) ([]string, error) {
+	isAdminCtx, err := portworx.IsAdminCtx(ctx)
+	if err != nil {
+		return nil, err
+	}
+	backupEnumerateReq := &api.BackupEnumerateRequest{
+		OrgId: orgID,
+		EnumerateOptions: &api.EnumerateOptions{
+			Owners: func() []string {
+				if isAdminCtx {
+					return []string{ownerID}
+				}
+				return nil
+			}(),
+		},
+	}
+	backupEnumerateResp, err := Inst().Backup.EnumerateBackup(ctx, backupEnumerateReq)
+	if err != nil {
+		return nil, err
+	}
+	backupNames := make([]string, 0)
+	for _, backupObj := range backupEnumerateResp.GetBackups() {
+		if isAdminCtx {
+			backupNames = append(backupNames, backupObj.GetName())
+		} else {
```

Comment:
why we need a else logic here, if its not admin context , There will be only respective user backup in enumerate resp.

Reply:
I am using these functions to validate both backup share and cluster backup share. If User 1 shares a backup with User 2, I can use the GetAllBackupNamesByOwnerID function, passing in owner ID of user 1 and user 2 (non-admin) ctx, to list the backup names and validate.

Reply:
Make sense, Thanks for clarification.